tests/e2e.rs

```diff
@@ -71,3 +69,54 @@ async fn e2e_channel_update() {
     assert_eq!(alice.balance().unwrap().ours, bob.balance().unwrap().theirs);
     assert_eq!(alice.balance().unwrap().theirs, bob.balance().unwrap().ours);
 }
+
+#[tokio::test]
+async fn e2e_punish_publication_of_revoked_commit_transaction() {
+    let tc_client = testcontainers::clients::Cli::default();
+    let bitcoind = Bitcoind::new(&tc_client, "0.19.1").unwrap();
+
+    let time_lock = 10;
+    let (alice_balance, bob_balance) = (Amount::ONE_BTC, Amount::ONE_BTC);
+
+    let CreateActors {
+        alice,
+        alice_wallet,
+        bob,
+        bob_wallet,
+    } = make_create_actors(&bitcoind, alice_balance, bob_balance, time_lock).await;
+
+    let Created { alice, bob } = run_create_protocol(&alice_wallet, alice, &bob_wallet, bob).await;
+
+    alice_wallet
+        .0
+        .send_raw_transaction(alice.signed_TX_f.clone())
+        .await
+        .unwrap();
+
+    let UpdateActors { alice, bob } = make_update_actors(alice, bob);
+
+    let channel_update = ChannelUpdate::Pay(Amount::from_btc(0.5).unwrap());
+    let time_lock = 10;
+
+    let Updated { alice, bob } = run_update_protocol(alice, bob, channel_update, time_lock);
+
+    // Alice attempts to cheat by publishing a revoked commit transaction
+
+    let signed_revoked_TX_c = alice.latest_revoked_signed_TX_c().unwrap().unwrap();
+    alice_wallet
+        .0
+        .send_raw_transaction(signed_revoked_TX_c.clone())
+        .await
+        .unwrap();
+
+    // Bob sees the transaction and punishes Alice
+
+    let bob = punish::Party0::from(bob);
+    let TX_p = bob.punish(signed_revoked_TX_c).unwrap();
```

Comment:
Best to assert onchain balances to ensure the test pass. I guess that `send_raw_transaction` being successful shows that "it works" but a balance test is in my opinion the ultimate goal.

Reply:
Yep, I tried to capture the limitation in the PR description.

I think for the time being we should just use the "punish output" as an input to a transaction which spends to a `bob_wallet`-owned address, sign it with `bob.x_self`, broadcast it and then check that bob gets paid.

Reply:
> Yep, I tried to capture the limitation in the PR description.

Oops, sorry, I reviewed this in several passes due to the hectic day.